bench_lib.rb

```diff
@@ -223,30 +235,30 @@ def out_file
     def run_wrk
       filename = @settings[:json_filename]
       File.open(filename, "w") { |f| f.write JSON.dump(@settings) }
-      exec_with_config @settings[:ruby_subprocess_cmd]
-      File.unlink(filename)
+      begin
+        exec_with_config @settings[:ruby_subprocess_cmd]
+      ensure
+        File.unlink(filename)
+      end
     end
 
     def exec_with_config(cmd_line)
-      child_pid = fork do
-        Bundler.with_clean_env do
-          ENV["RACK_ENV"] = @settings[:rack_env]
-          ENV["RAILS_ENV"] = @settings[:rack_env]
-          if @settings[:bundle_gemfile]
-            ENV["BUNDLE_GEMFILE"] = @settings[:bundle_gemfile]
-          end
-          if @settings[:bundler_version]
-            ENV["BUNDLER_VERSION"] = @settings[:bundler_version]
-          end
-          if @settings[:extra_env]
-            @settings[:extra_env].each { |k, v| ENV[k.to_s] = v.to_s }
-          end
-          verbose "exec: #{cmd_line.inspect}"
-          exec cmd_line
+      Bundler.with_clean_env do
+        env = {}
+        env["RACK_ENV"] = @settings[:rack_env]
+        env["RAILS_ENV"] = @settings[:rack_env]
+        if @settings[:bundle_gemfile]
+          env["BUNDLE_GEMFILE"] = @settings[:bundle_gemfile]
+        end
+        if @settings[:bundler_version]
+          env["BUNDLER_VERSION"] = @settings[:bundler_version]
+        end
+        if @settings[:extra_env]
+          @settings[:extra_env].each { |k, v| env[k.to_s] = v.to_s }
         end
+        verbose "exec: #{env.map { |k,v| "#{k}=#{v}" }.join(' ')} #{cmd_line}"
+        system env, cmd_line
```

Comment:
I really like this way of handling the environment. I did something similar in a branch, though this is better yet. 👍